parlai/scripts/interactive_web.py

```diff
@@ -0,0 +1,280 @@
+                    article.appendChild(figure);
+                    article.appendChild(media);
+
+                    parDiv.append(article);
+
+                    // Change info for Model response
+
+                    var articleM = document.createElement("article");
+                    articleM.className = "media";
+
+                    var figureM = document.createElement("figure");
+                    figureM.className = "media-left";
+
+                    var spanM = document.createElement("span");
+                    spanM.className = "icon is-large";
+
+                    var iconM = document.createElement("i");
+                    iconM.className = "fas fa-robot fas fa-2x";
+
+                    var mediaM = document.createElement("div");
+                    mediaM.className = "media-content";
+
+                    var contentM = document.createElement("div");
+                    contentM.className = "content";
+
+                    var paraM = document.createElement("p");
+                    var paraTextM = document.createTextNode(data.text);
+                    console.log('paraTextM');
+                    console.log(paraTextM);
+
+                    var strongM = document.createElement("strong");
+                    strongM.innerHTML = "Model";
+                    var brM = document.createElement("br");
+
+                    paraM.appendChild(strongM);
+                    paraM.appendChild(brM);
+                    paraM.appendChild(paraTextM);
+                    contentM.appendChild(paraM);
+                    mediaM.appendChild(contentM);
+
+                    spanM.appendChild(iconM);
+                    figureM.appendChild(spanM);
+
+                    articleM.appendChild(figureM);
+                    articleM.appendChild(mediaM);
+
+                    parDiv.append(articleM);
+                }})
+            }});
+        </script>
+
+    </body>
+</html>
+"""  # noqa: E501
+
+
+class MyHandler(BaseHTTPRequestHandler):
+
+    def interactive_running(self, opt, reply_text):
+        reply = {}
+        reply['text'] = reply_text
+        SHARED['agent'].observe(reply)
+        model_res = SHARED['agent'].act()
+        return model_res
+
+    def do_HEAD(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+
+    def do_POST(self):
+        if self.path == '/interact':
+            content_length = int(self.headers['Content-Length'])
+            body = self.rfile.read(content_length)
+            model_response = self.interactive_running(
+                SHARED.get('opt'),
+                body.decode('utf-8')
+            )
+
+            self.send_response(200)
+            self.send_header('Content-type', 'application/json')
+            self.end_headers()
+            json_str = json.dumps(model_response)
+            self.wfile.write(bytes(json_str, 'utf-8'))
+
+        else:
+            self.respond({'status': 500})
```

Comment:
One way you can have cleaner code in such a situation is invert the condition and fail fast.

```
if self.path != '/interact':
   return self.respond({'status': 500})

content_length = int(self.headers['Content-Length'])
...
```